shared-bindings/usb_cdc/Serial.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 Dan Halbert for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <stdint.h>
+
+#include "shared-bindings/usb_cdc/Serial.h"
+#include "shared-bindings/util.h"
+
+#include "py/ioctl.h"
+#include "py/objproperty.h"
+#include "py/runtime.h"
+#include "py/stream.h"
+#include "supervisor/shared/translate.h"
+
+//| class Serial:
+//|     """Receives cdc commands over USB"""
+//|
+//|     def __init__(self) -> None:
+//|         """You cannot create an instance of `usb_cdc.Serial`.
+//|
+//|         Serial objects are pre-constructed for each CDC device in the USB
+//|         descriptor and added to the ``usb_cdc.ports`` tuple."""
+//|         ...
+//|
+
+//|     def read(self, size: int = 1) -> bytes:
+//|         """Read at most ``size`` bytes. If ``size`` exceeds the internal buffer size
+//|         only the bytes in the buffer will be read. If `timeout` is > 0 or ``None``,
+//|         and fewer than ``size`` bytes are available, keep waiting until the timeout
+//|         expires or ``size`` bytes are available.
+//|
+//|         :return: Data read
+//|         :rtype: bytes"""
+//|         ...
+//|
+//|     def readinto(self, buf: WriteableBuffer) -> bytes:
```

Comment:
```suggestion
//|     def readinto(self, buf: WriteableBuffer) -> int:
```

Doesn't usually return the count of bytes filled in?

Reply:
Good catch, that's just a typo.